Origin: http://hg.intevation.org/mercurial/crew/rev/f2b1f78cf202
Bug: http://bugs.debian.org/701168
Subject: mergetools: vimdiff issue a warning explaining how to abort

Adds a message displayed at each vimdiff invocation:

  merge conflict detected, type ":cq" to abort

Vimdiff is very confusing for non-vim user (not to speak about vim
user confused anyway. However it is very likely that vimdiff is picked
as the mergetool of choice when using the default config:
- vim is available on all UNIX system.
- Its one of the rare non graphical merge tools.

diff --git a/contrib/mergetools.hgrc b/contrib/mergetools.hgrc
--- a/contrib/mergetools.hgrc
+++ b/contrib/mergetools.hgrc
@@ -15,7 +15,7 @@ gvimdiff.regkeyalt=Software\Wow6432Node\
 gvimdiff.regname=path
 gvimdiff.priority=-9
 
-vimdiff.args=$local $other $base
+vimdiff.args=$local $other $base -c 'echohl WarningMsg | echo "merge conflict detected, type \":cq\" to abort" | echohl'
 vimdiff.check=changed
 vimdiff.priority=-10
 

